### DISCUSSION TITLE
REopt issue

### DISCUSSION THREAD

**Discussion by aisyah-zakiah** [urbanopt/urbanopt-cli#470] [Q&A]
Hi!

I'm trying to run REopt using the example file.

`uo process -e -s Project_1_reopt/REopt_baseline_scenario.csv -f Project_1_reopt/example_project_with_PV.json`

However, I'm encountering this issue:

> Post-processing each building individually with REopt
> E, [2024-06-20T15:02:21.658423 #9800] ERROR -- : make_request Failed
> E, [2024-06-20T15:02:21.658645 #9800] ERROR -- : ERROR: REopt connection Failed

Do you know what could be wrong?

![image](https://github.com/urbanopt/urbanopt-cli/assets/173452456/7a17a46b-b83d-49a8-be9f-297faab22853)

**Comment by kflemin** [comment]
hi @aisyah-zakiah, a couple of things for you to check:
1. Have you followed the directions to set up your REopt API key?
2. Look at issue number 2 in this page: https://docs.urbanopt.net/developer_resources/known_issues.html.  If you are using the installer, follow the directions to location your "certified-update" path and run it.  

Let me know how it goes!

**Reply by aisyah-zakiah** [reply]
hi @kflemin ,

1. I have set the API key in the environment variables.
2. Certified-update has been successfully executed.

However, I'm still facing the "REopt connection failed" error.
I would appreciate any further insights you might have.

![image](https://github.com/urbanopt/urbanopt-cli/assets/173452456/f159af45-ad80-4ad0-825d-dd2b6984ce1b)

**Comment by xyl010203** [comment]
At first, I had the same problem as you, which was that the make request failed.Then I updated the certificate by running the certified-update file in the URBANopt-cli-0.13.0\gems\ruby\2.7.0\gems\certified-1.0.0\bin directory. Now it can run, but it reports an error with the code 400. I have set up proxies for both http and https on Git Bash to access web resources through the port of Clash. The error message is as follows.

"""Post-processing entire scenario with REopt
W, [2024-10-29T21:25:55.658066 #16120]  WARN -- : REopt has returned a '400' status code. Visit https://developer.nrel.gov/docs/errors/ for more status code information
E, [2024-10-29T21:25:55.661562 #16120] ERROR -- : MESSAGES: {"errors"=>{"core_urdb.jl_134"=>[" Error occurred HTTP.Exceptions.StatusError(500", " GET", " utility_ratesapi_key2qt5uihpKXdywTj3uMIhBewxY9K4eNjpRje1JUPLversion8formatjsondetailfullgetpage63d2f36655296095a0092a6e", " HTTP.Messages.ResponsennHTTP1.1 500 Internal Server ErrorrnDate Tue", " 29 Oct 2024 132543 GMTrnContentType texthtml charsetUTF8rnContentLength 0rnConnection keepalivernAccessControlAllowOrigin rnAge 131rnExpectCt maxage7776000", " reporturihttpsopenei.orgwikiOpenEIAboutrnStrictTransportSecurity maxage31536000preload", " maxage31536000 preloadrnVia ma4", " fs8.openei.org", " https1.1 apiumbrella (ApacheTrafficServer cMsSf )rnXApiUmbrellaRequestId cir1p2aube77vjkft940rnXCache MISS from fs8.openei.orgrnXContentTypeOptions nosniffrnXRatelimitLimit 1000rnXRatelimitRemaining 984rnXVcapRequestId 6709a4972c33417850dddad5dcf40347rnXXssProtection 1 modeblockrnXFrameOptions sameoriginrnrn)"]}, "has_stacktrace"=>false}
response.code: 400
message: Bad Request
"""

**Comment by kflemin** [comment]
We have recently released newer versions of URBANopt which addresses many prior issues. If you are still experiencing issues related to this after upgrading, please let us know.
